sulong/projects/com.oracle.truffle.llvm.test/src/com/oracle/truffle/llvm/test/debug/LLVMDebugExprParserTest.java

```diff
@@ -0,0 +1,236 @@
+        final Path path = TRACE_DIR_PATH.resolve(testName + TRACE_EXT);
+        return TestExpressions.parse(path);
+    }
+
+    private static final class BreakInfo {
+
+        private int lastStop;
+        private ContinueStrategy lastStrategy;
+
+        BreakInfo() {
+            this.lastStop = -1;
+            this.lastStrategy = null;
+        }
+
+        int getLastStop() {
+            return lastStop;
+        }
+
+        void setLastStop(int lastStop) {
+            this.lastStop = lastStop;
+        }
+
+        ContinueStrategy getLastStrategy() {
+            return lastStrategy;
+        }
+
+        void setLastStrategy(ContinueStrategy lastStrategy) {
+            this.lastStrategy = lastStrategy;
+        }
+    }
+
+    private static final class TestCallback implements SuspendedCallback {
+
+        private final BreakInfo info;
+        private final StopRequest bpr;
+        private final Map<String, String> textExpressionMap;
+
+        TestCallback(BreakInfo info, StopRequest bpr, Map<String, String> textExpressionMap) {
+            this.info = info;
+            this.bpr = bpr;
+            this.textExpressionMap = textExpressionMap;
+        }
+
+        private static void setStrategy(SuspendedEvent event, DebugStackFrame frame, ContinueStrategy strategy) {
+            if (strategy != null) {
+                strategy.prepareInEvent(event, frame);
+            } else {
+                ContinueStrategy.STEP_INTO.prepareInEvent(event, frame);
+            }
+        }
+
+        @Override
+        public void onSuspend(SuspendedEvent event) {
+            final DebugStackFrame frame = event.getTopStackFrame();
+
+            final int currentLine = event.getSourceSection().getStartLine();
+
+            if (currentLine == info.getLastStop()) {
+                // since we are stepping on IR-instructions rather than source-statements it can
+                // happen that we step at the same line multiple times, so we simply try the last
+                // action again. The exact stops differ between LLVM versions and optimization
+                // levels which would make it difficult to record an exact trace.
+                setStrategy(event, frame, info.getLastStrategy());
+                return;
+
+            } else if (currentLine == bpr.getLine()) {
+                info.setLastStop(currentLine);
+                info.setLastStrategy(bpr.getNextAction());
+                setStrategy(event, frame, bpr.getNextAction());
+
+            } else {
+                throw new AssertionError(String.format("Unexpected stop at line %d", currentLine));
+            }
+
+            for (Entry<String, String> kv : textExpressionMap.entrySet()) {
+                String actual = frame.eval(kv.getKey()).as(String.class);
+                System.out.println("\"" + kv.getValue() + "\" vs. \"" + actual + "\"");
+                assertEquals("Expressions are not equal", kv.getValue(), actual);
```

Comment:
This error message should rather read something like `Evaluation of expression 'xyz' produced unexpected result`.